apps/web/src/app/HSL-API-endpoint/fetch-stops/route.ts

```diff
@@ -0,0 +1,210 @@
+import { ApolloClient, gql, InMemoryCache } from "@apollo/client";
+import type { ApolloQueryResult, DocumentNode } from "@apollo/client";
+import type {
+  Arrival,
+  ArrivalAttribute,
+  RenderableStop,
+  Stop,
+  StopOutData,
+  StopTime,
+  StopType,
+  Data,
+  StopFromApi,
+} from "../../../assets/HSL-API-types/hsl-helper-types";
+
+const STOPS = [
+  ["HSL:2222406", "HSL:2222405"],
+  ["HSL:2222603", "HSL:2222604"],
+  ["HSL:2222234", "HSL:2222212"],
+]; // [["HSL:2222234", "HSL:E22259"], ["HSL:2000102"], ["HSL:E0773", "HSL:E0772"]];
+
+export const dynamic = "force-dynamic";
+
+export async function GET() {
+  const dataFromHsl: RenderableStop[] = [];
+  for (const stop of STOPS) {
+    const stp: RenderableStop | null = await getStop(stop);
```

Comment:
Do these need to be done sequentially? Or could we parallelize the requests for blazing fast speed 🔥

Reply:
It could be done but it fetches 10 arrivals for 6 stops in 42ms in localhost so I think the benefit of parallellism would be marginal but if it slows down significantly in production it can be implemented later.

Reply:
There is no reason here not to use [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)/[Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) with a map inside instead of a for loop, don't know why ESLint is not complaining about the `await` inside the loop, I guess we have that rule disabled?